app/js/services/MutationsService.js

```diff
@@ -32,7 +32,7 @@ function MutationsService(
 
     // promise wrapper
     // @todo: handle reject condition
-    return new Promise(resolve => {
+    return $q(resolve => {
```

Comment:
Good catch.

So, in this case, is there a corresponding `$scope.$apply()` we need to call? It looks as if the code that invokes this `.query()` method is `queryParamSelector.js` at line 122, and I don't see any `$apply()` to clean up there, but I want to be sure we don't miss one or we'd be introducing a new bug with this change...

Reply:
Just did a quick scan for `$apply(` in the codebase, and I see only one that's not coming from an Angular framework... at line 19 of `app.component.js`. That one appears to arise from `UserAuth.login()`. Sure enough, `UserAuth.js` returns a non-angular Promise in three spots. Maybe that should be a separate fix, but maybe it's also not broken since it *does* have the `$apply()` in there...

Reply:
Hah... here's a better way to search: look through the whole codebase for `Promise(`. That brings up this one we're fixing (`MutationsService.js`), what I spotted in `UserAuth.js` above, and three more: `js/services/ProgressIndicatorBarService.js`, `gulp/util/testServer.js` (probably not an issue for us), and `js/services/example.js`.